shared/chat/blocking/block-modal/index.tsx

```diff
@@ -0,0 +1,283 @@
+  onCheck: (boolean) => void
+  text: React.ReactNode
+}
+const CheckboxRow = (props: CheckboxRowProps) => {
+  const [infoShowing, setInfoShowing] = React.useState(false)
+  return (
+    <>
+      <Kb.Box2 direction="horizontal" alignItems="center" fullWidth={true} style={styles.checkBoxRow}>
+        <Kb.Switch
+          color="red"
+          gapSize={Styles.globalMargins.tiny}
+          label={props.text}
+          labelSubtitle={infoShowing ? props.info : undefined}
+          on={props.checked}
+          onClick={() => props.onCheck(!props.checked)}
+          style={styles.shrink}
+        />
+        <Kb.Box style={styles.iconBox} />
+        {props.info && !infoShowing && (
+          <Kb.Icon type="iconfont-question-mark" color="grey" onClick={() => setInfoShowing(true)} />
+        )}
+      </Kb.Box2>
+    </>
+  )
+}
+
+type ReportOptionsProps = {
+  extraNotes: string
+  includeTranscript: boolean
+  reason: string
+  setExtraNotes: (text: string) => void
+  setIncludeTranscript: (checked: boolean) => void
+  setReason: (reason: string) => void
+  showIncludeTranscript: boolean
+}
+const reasons = ["I don't know this person", 'Spam', 'Harassment', 'Obscene material', 'Other...']
+const ReportOptions = (props: ReportOptionsProps) => {
+  return (
+    <>
+      {reasons.map(reason => (
+        <Kb.RadioButton
+          key={reason}
+          label={reason}
+          onSelect={() => props.setReason(reason)}
+          selected={props.reason === reason}
+          style={styles.radioButton}
+        />
+      ))}
+      <Kb.Box style={styles.feedback}>
+        <Kb.NewInput
+          multiline={true}
+          placeholder="Extra notes"
+          onChangeText={text => props.setExtraNotes(text)}
+          value={props.extraNotes}
+        />
+      </Kb.Box>
+      {props.showIncludeTranscript && (
+        <CheckboxRow
+          text="Include the transcript of this chat"
+          onCheck={checked => props.setIncludeTranscript(checked)}
+          checked={props.includeTranscript}
+        />
+      )}
+    </>
+  )
+}
+
+class BlockModal extends React.PureComponent<Props, State> {
+  state = {
+    blockTeam: true,
+    extraNotes: '',
+    finishClicked: false,
+    includeTranscript: false,
+    // newBlocks holds a Map of blocks that will be applied when user clicks
+    // "Finish" button.
+    newBlocks: new Map(),
+    reportReason: reasons[0],
+    shouldReport: false,
+  }
+
+  componentDidMount() {
+    // Once we get here, trigger actions to refresh current block state of
+    // users.
+    this.props.refreshBlocks()
+
+    // Set default checkbox block values for adder user. We don't care if they
+    // are already blocked, setting a block is idempotent.
+    if (this.props.blockByDefault) {
+      const map = this.state.newBlocks
+      map.set(this.props.adderUsername, {chatBlocked: true, followBlocked: true})
+      this.setState({newBlocks: new Map(map)})
+    }
+  }
+
+  setReport(checked: boolean) {
```

Comment:
prefer `private setReport = (checked: boolean) =>` for the non-lifecycle methods